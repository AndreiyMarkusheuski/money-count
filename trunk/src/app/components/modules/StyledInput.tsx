import React from "react";

type Props = {
    typeInput: string,
    placeholder: string,
    valueInput: string,
    handleChange: (e: any) => void
}

const StyledInput = ({ typeInput, placeholder, valueInput, handleChange }: Props) => {
    return (
        <input
            type={typeInput}
            placeholder={placeholder}
            value={valueInput}
            onChange={handleChange}
            data-testid="add-word-input"
        ></input>
    )
}

export default StyledInput;