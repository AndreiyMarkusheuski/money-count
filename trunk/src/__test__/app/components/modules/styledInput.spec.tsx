import React from "react";
import { render, RenderResult, fireEvent } from '@testing-library/react';
import StyledInput from '../../../../app/components/modules/StyledInput'

let documentBody: RenderResult;

describe('<StyledInput/>', () => {
    const testData = {
        typeInput: 'number',
        placeholder: 'placeholder',
        valueInput: '250',
        handleChange: () => { console.log('test') }
    }

    beforeEach(() => {
        documentBody = render(<StyledInput {...testData} />);
    });

    it('match snapshot', () => {
        const { baseElement } = documentBody;
        expect(baseElement).toMatchSnapshot();
    })

    it('should render the component onto the screen', () => {
        expect(documentBody.getByTestId('add-word-input')).toBeInTheDocument();
    })

    it('should render the component onto the screen with value "test" and placeholder "placeholder"', () => {
        const input = documentBody.getByTestId('add-word-input') as HTMLInputElement;
        expect(input.value).toBe('250')
    })

    it('should render the component onto the screen and update value', () => {
        const input = documentBody.getByTestId('add-word-input') as HTMLInputElement;

        fireEvent.click(input, {
            target: {
                value: "23"
            }
        })

        expect(input.value).toBe('23')
    })
})