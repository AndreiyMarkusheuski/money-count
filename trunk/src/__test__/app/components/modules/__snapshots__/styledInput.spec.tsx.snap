// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledInput/> match snapshot 1`] = `
<body>
  <div>
    <input
      data-testid="add-word-input"
      placeholder="placeholder"
      type="number"
      value="250"
    />
  </div>
</body>
`;
